### PR TITLE
Remove Relic submodule from the repo

### PR DESCRIPTION
**Pull request by tarakby**
This PR removes the Relic submodule from the flow-go repo. In order to set up `flow-go/crypto` package, running `go generate` from the package directory is enough: 

- This is a step further towards having a self-sufficient `flow-go/crypto` package that doesn't depend on the Github flow-go repo to track the Relic version needed. 
- Setting up `flow-go/crypto` still requires a step managed by a script to build a static library for Relic (nothing changed here). With this PR, the script step includes cloning the required commit version from the relic repo and is triggered by the go tool `go generate`. 
- Removing submodules from the repo avoids accidentals commits of local submodules by contributors (including the Flow team). 
- For flow-go developers, cloning Relic files and history locally may take a few seconds (about 2s on my laptop) but this should only happen when Relic version is updated, or when switching between branches with a different relic version. 
- This PR also fixes the name of the main branch to clone relic that got updated from `master` to `main`
- add @huitseeker as a crypto code owner to help keep an eye on future updates of the package.

### CODE REVIEW

File: crypto/build_dependency.sh
Comment (huitseeker) [inline crypto/build_dependency.sh:13]: I would use the long SHA1, as this line has suddenly become a very important part of the build (granted, SHA1 is broken all on its own 🤦 )

File: crypto/README.md
Comment (huitseeker) [inline crypto/README.md:29]: We should probably document the bash requirement as well (unless and until we replace the build-dependency script with a go program).

**Review by huitseeker — COMMENTED** [review]
I'll assume this is WIP, and mention upcoming high-level concerns rather than missing code bits:

How do we address forensics? I.e. if I compile once, pull a commit that changes the pointer to the relic version, recompile. How do we make sure I (as a developer) can find out that I either:
- forgot to run `go generate` or,
- that the Makefile considered the presence of files output by `go generate` to not be sufficient to satisfy the corresponding goal and re-ran it?[1]

[1]: that might point at a Makefile phony goal that inspects the value for the commit in the build-dependency script and re-runs the `go generate` goal if changed ... which is starting to look like https://github.com/onflow/flow-go/pull/974

Do we have numbers on the time cost of pulling the relic Github checkout every time?

Does this address the cache invalidation of the [Docker build's copy of the go build cache](https://github.com/onflow/flow-go/blob/master/cmd/Dockerfile#L45-L51)?

**Comment by tarakby** [comment]
> How do we make sure I (as a developer) can find out that I either:
forgot to run go generate

I do not have a good solution for this without passing by the flow-go makefile. If a developer only gets the crypto package and not flow-go, they will need to run `go generate`.  

> that might point at a Makefile PHONY goal that inspects the value for the commit in the build-dependency script and re-runs the go generate goal if changed ... which is starting to look like #974

For developers working on flow-go, I still think #974 check is useful. We might just need to update it to make it compare the version hash in `build_dependency.sh` against the git hash of crypto/relic (something like `git rev-parse HEAD`). 

> I'll assume this is WIP, and mention upcoming high-level concerns rather than missing code bits
Yep I'll open it and request your review for the full code review 👌

**Comment by tarakby** [comment]
> Do we have numbers on the time cost of pulling the relic Github checkout every time?

It takes about 2s on my laptop, but I agree it's not ideal to clone the entire history just for a specific commit version. I didn't find a way for git to pull a specific commit unless it is a branch or a tag. 
For developers, the relic clone should only happen when cloning a fresh flow-go (which is long anyway) or when updating relic version (which doesn't happen often). 

> Does this address the cache invalidation of the Docker build's copy of the go build cache?

No I haven't tried to address that.

File: .github/workflows/ci.yml
Comment (tarakby) [inline .github/workflows/ci.yml:51]: To be removed before merging

File: Makefile
Comment (tarakby) [inline Makefile:141]: Having the check in `tidy` make `make ci` fail since `install-tools` and `crypto/relic/check` are run in parallel. The crypto check fails to read the version from the relic directory because it wasn't created yet.

I suggest to leave `crypto/relic/check` for local development checks only. On CI. there is no risk to have mistakes since there is only one source of truth for the relic version.

File: crypto/README.md
Comment (tarakby) [reply crypto/README.md:29]: Do you mean requirements as in the version?

**Review by huitseeker — APPROVED** [review]
Accepting to unblock: On the plus side the repo is now dependent on `go generate` and `git clone` rather than a submodule, but I feel that if a developer is e.g. using an IDE (that is, not re-running the build through the Makefile), a new commit changing the relic version may not be picked up before a PR & running CI.

I left a few nits in the file, the main being:
- removing the tmate session,
- adding a CODEOWNERS file to ensure review of the relic version.

File: crypto/build_dependency.sh
Comment (tarakby) [reply crypto/build_dependency.sh:13]: Do you mind if I add you as codeowner for `/crypto/` ?

Comment (huitseeker) [reply crypto/build_dependency.sh:13]: Nope!

File: integration/loader/Dockerfile
Comment (tarakby) [reply integration/loader/Dockerfile:10]: Can the suggested command be applied only for `git clone relic` and we could include it in `build_dependency.sh` instead of the docker file?

Comment (huitseeker) [reply integration/loader/Dockerfile:10]: Yep, good idea, I believe that's option `-c`

File: crypto/build_dependency.sh
Comment (huitseeker) [inline crypto/build_dependency.sh:5]: we could use the more reliable:
`DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"`

File: crypto/build_dependency.sh
Comment (huitseeker) [inline crypto/build_dependency.sh:21]:
```suggestion
git -c http.sslVerify=true clone --branch main --single-branch https://github.com/relic-toolkit/relic.git relic || { echo "git clone failed"; exit 1; }
```

This protects against repo renamings

**Review by huitseeker — COMMENTED** [review]
(nits)

File: Makefile
Comment (AlexHentschel) [inline Makefile:77]: did anything change here except a blank space at the line end (?)

**Review by AlexHentschel — APPROVED** [review]
LGTM

File: Makefile
Comment (tarakby) [reply Makefile:77]: No, it must be a left over from debugging :)

File: crypto/build_dependency.sh
Comment (tarakby) [reply crypto/build_dependency.sh:21]: 💡

**Pull request merged**
